imageutils.py

```diff
@@ -246,17 +251,19 @@ def split_clustered_writes(self, Y_splits, Z_splits, X_splits, out_dir, mem, fil
                 x_e = split_start[2] + x_size
 
                 split_data = data[split_start[0] : y_e, split_start[1] : z_e, split_start[2] : x_e]
-                im = nib.Nifti1Image(split_data, self.affine)
+                # im = nib.Nifti1Image(split_data, self.affine)
                 
                 t = time()
-                nib.save(im, split_names[start_index + j])
+
+                write_array_to_file(split_data, split_names[start_index + j], write_offset=  self.header_size, hdfs_client=hdfs_client)
+                # nib.save(im, split_names[start_index + j])
```

Comment:
remove comment.

Reply:
So i believe i have figured out how to get nibabel objects into a bytestream such that we can use nibabel and write directly to HDFS (and also useful for Spark RDDs). I tested this out quickly and did no real byte comparison, so it might be a good idea to double check @YongpingGao, should you choose to use this

Basically:
```
im = nib.load(<filename>)
bytestream = im.file_map['image'].get_prepare_fileobj('rb') # i just had an image fileholder, but there can be a header one too if the image is split (.hdr, .img), i think
```
I think it's fine though to have switched it as it's good to be consistent throughout the different algorithms, but just throwing this out there in case you could use it.

Reply:
Yeah, that seems a good idea, I will double check its performance. And this way I think we don't have to bind to one specific hdfs lib in python.